src/Utilities/lazy_iterators.jl

```diff
@@ -0,0 +1,65 @@
+struct EmptyVector{T} <: AbstractVector{T} end
+Base.size(::EmptyVector) = (0,)
+Base.isempty(::EmptyVector) = true
+Base.eltype(::EmptyVector{T}) where {T} = T
+Base.iterate(::EmptyVector) = nothing
+
+struct LazyMap{T, VT}
```

Comment:
This could be replaced by `Base.Generator(f, data)`. See https://discourse.julialang.org/t/why-no-base-iterators-map/14280/7.

Reply:
The difference is that `eltype(Base.Generator(x -> x^2, 1:2))` while for `LazyMap`, you need to set the eltype

Reply:
Got it. Please add a comment explaining.

Reply:
In fact, this is a simplified version of https://github.com/JuliaArrays/MappedArrays.jl